services/common/rs_server_common/authentication.py

```diff
@@ -120,3 +148,67 @@ async def __apikey_security_cached(apikey_value) -> tuple[list, dict]:
 
     # Forward error
     raise HTTPException(response.status_code, f"UAC manager: {detail}")
+
+
+def apikey_validator(station, access_type):
+    """Decorator to validate API key access.
+
+    Args:
+        station (str): The station name.
+        access_type (str): The type of access.
+
+    Raises:
+        HTTPException: If the authorization key does not include the right role
+            to access the specified station.
+
+    Returns:
+        function: Decorator function.
+    """
+
+    def decorator(func):
+        @wraps(func)
+        def wrapper(*args, **kwargs):
+            if settings.cluster_mode():
+                try:
+                    __station = STATIONS_AUTH_lut[kwargs["station"].lower()] if station == "cadip" else station
```

Comment:
Why not directly __station = STATIONS_AUTH_lut[kwargs["station"].lower()] since "adgs" is already in LUT